cmake/modules/BuildBoost.cmake

```diff
@@ -158,6 +168,12 @@ function(do_build_boost version)
     PREFIX "${boost_root_dir}")
 endfunction()
 
+set(Boost_context_DEPENDENCIES thread chrono system date_time)
+set(Boost_coroutine_DEPENDENCIES context system)
+set(Boost_filesystem_DEPENDENCIES system)
+set(Boost_iostreams_DEPENDENCIES regex)
+set(Boost_thread_DEPENDENCIES chrono system date_time atomic)
```

Comment:
I'm confused by the presence of "chrono" here, because when I look at Shaman build logs, the Boost::chrono module shows up as "not building"

See e.g. https://jenkins.ceph.com/job/ceph-dev-build/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=centos7,DIST=centos7,MACHINE_SIZE=huge/15202//consoleFull (grep for "chrono")

Reply:
The reason I'm investigating this: s390x builds started to fail with

```
[ 7791s] Scanning dependencies of target ceph-common
[ 7792s] make[2]: *** No rule to make target 'boost/lib/libboost_chrono.a', needed by 'lib/libceph-common.so.0'.  Stop.
[ 7792s] make[1]: *** [CMakeFiles/Makefile2:1047: src/CMakeFiles/ceph-common.dir/all] Error 2
```

Reply:
see https://github.com/ceph/ceph/pull/20977 for a possible fix